app/app.go

```diff
@@ -29,6 +30,11 @@ func SetupRoutes(app_settings common.AppSettings, database database.Database) *g
 	// DO not cache as it needs to handlenew form values
 	r.POST("/contact-send", makeContactFormHandler())
 
+	// this will setup the route for pagination
+	r.GET("/page/:num", func(c *gin.Context) {
```

Comment:
Let's add this endpoint as a cache-able one here. You can do something simlar to that's above in this file:

```go
addCachableHandler(r, "GET", "/page/:num", homeHandler, &cache, app_settings, database)
```

Also, calling the `homeHandler(c, app_settings, database)` here won't actually serve the HTML contents, because this function will only return the HTML buffer. Adding it with the cache wrapper will take care of that for you.